### PR TITLE
DBZ-4303 Add optionally enabled Debezium KC REST extension to tagged KC container image

### PR DESCRIPTION
**Pull request by dude0001**

### CODE REVIEW

File: connect/2.1/Dockerfile
Comment (dude0001) [inline connect/2.1/Dockerfile:17]: Looking for suggestions on how to move forward here. I do not see a DBZ v2.1 artifact for the KC REST Extension at https://repo1.maven.org/maven2/io/debezium/debezium-connect-rest-extension/ and this is from the v2.0.0 artifact (https://repo1.maven.org/maven2/io/debezium/debezium-connect-rest-extension/2.0.0.Final/debezium-connect-rest-extension-2.0.0.Final.jar.md5). This will cause the build to fail as the scripts are expecting an artifact that matches the DBZ version ($DEBEZIUM_VERSION) of the image being built.

File: connect-base/2.1/README.md
Comment (dude0001) [inline connect-base/2.1/README.md:119]: Is there anything else I can add here?

File: connect/2.1/Dockerfile
Comment (dude0001) [reply connect/2.1/Dockerfile:17]: Nevermind, the variable is still pointing to v2.0 as a 2.1 release hasn't been cut yet I suppose.

File: connect-base/2.1/docker-entrypoint.sh
Comment (jpechane) [inline connect-base/2.1/docker-entrypoint.sh:98]: @dude0001 Looking at the script, would it be possible to refactor it a bit - using function maybe as it seems doing Apicurio, rest-extension and scripting is same in the functionality

**Comment by jpechane** [comment]
@dude0001 Thanks for the PR, I've left a samll comment.

Before merging this PR we need to prepare an update in release pipeline - https://github.com/debezium/debezium/blob/main/jenkins-jobs/pipelines/release-pipeline.groovy#L489

File: connect/2.1/Dockerfile.local
Comment (jcechace) [inline connect/2.1/Dockerfile.local:18]: I think the correct extension here is ``jar`` as running the assembly profile locally doesn't seem to produce any tar archives for debezium-connect-rest-extension module.

Comment (dude0001) [reply connect/2.1/Dockerfile.local:18]: Yes, great catch. I didn't realize there wasn't a .tar.gz file for this artifact. I have a change locally for this for the Maven download as well. I'm working on Jiri's request to commonize some logic and will have those changes up shortly. I was running into problems testing fully with buildx yesterday because of self-signed certificates, but I've got that working now as well so doing more testing as well.

```
docker-maven-download central io/debezium debezium-connect-rest-extension "$DEBEZIUM_VERSION" "$KCRESTEXT_MD5" && \
```

Comment (jcechace) [reply connect/2.1/Dockerfile.local:18]: @dude0001  what issue with buildx? I can't think of anything where certificates should be an issue. I'm currently running the build for arm to verify if the failure in GH actions was a temporary glitch or if there is some issue with npm installation again. 

If you can fix the extension I will verify the full functionality as I have the change to release pipeline ready.

Comment (dude0001) [reply connect/2.1/Dockerfile.local:18]: It is an issue in my personal development I was able to work through. We have some software that inspects all SSL traffic on the network that many CLI tools see rightly as a man-in-the-middle attack. I have to hack the dockerfiles in my local development environment so Docker, the OS in the Docker containers and CLI tools running in Docker containers trust the self-signed certs on my network used for this purpose. The product is Netskope but there are a few other tools that security organizations deploy internally to enterprises these days that is a nightmare for developers. I usually am able to work around it fine but was having an issue with buildx itself and how it pulls images from registries, but finally figured it out this morning and was able to reconfigure my local environment for it. No issue with Debezium or this repo.

File: connect/2.1/Dockerfile
Comment (jcechace) [inline connect/2.1/Dockerfile:27]: This will fail as debezium-optional method assumes the target artifact is a tar archive

https://github.com/debezium/container-images/blob/main/connect-base/2.1/docker-maven-download.sh#L55


@jpechane  just to confirm, should there be a tar for connect-rest-extension?

Comment (jpechane) [reply connect/2.1/Dockerfile:27]: tar was no planned but thinking about that let's do it in the same way as `debezium-scripting` so it should provide tar too.

Comment (jcechace) [reply connect/2.1/Dockerfile:27]: @dude0001   based on Jiri's comment we should keep this and the extension in ``Dockerfile.local`` the way it is now. Tomorrow I will add the appropriate profile to the rest-extension module in main repository.

File: connect-base/2.1/docker-entrypoint.sh
Comment (dude0001) [reply connect-base/2.1/docker-entrypoint.sh]: The debezium-scripting and apicurio libraries are being added as additional resources to each Debezium Connector. The Debezium REST API Extension needs to be deployed as its own Kafka Connect plugin is my understanding, and not a resource for Debezium Connectors. If that statement is true, Debezium REST API Extension needs its own pattern.

I refactored out a function with shared logic for enabling additional resources to all Debezium Connectors, and debezium-scripting and apicurio use this now.

I created another function that is specific to the REST API Extension for now, which could be tweaked more in the future to enable other optional true Kafka Connect plugins.

I was able to test to be working for the REST API Extension and did some regression for debezium-scripting and apicurio, and it looks to be configuring everything as expected. Please take a look, I would love additional feedback if I need to take another pass at this.

File: connect/2.1/Dockerfile.local
Comment (dude0001) [reply connect/2.1/Dockerfile.local:18]: I've changed everything to assume a .tar.gz artifact will be available. For local testing, I hacked something to simulate that and I think it will work once that artifact is available. Please have another look.

File: connect/2.1/Dockerfile
Comment (dude0001) [reply connect/2.1/Dockerfile]: I refactor my logic including assuming there will be a .tar.gz artifact available for the REST API Extension. Please take another look.

**Comment by jpechane** [comment]
@dude0001 LGTM, thanks!
So we need to

1. Merge PR that would generate `tar.gz` for rest extension
2. Release Alpha1
3. Merge this PR
4. 2.1 tag will contain REST extension
5. We need to consider if we republish 2.1.0.Alpha1 tag with extension too

**Comment by dude0001** [comment]
Okay my understanding is this is blocked for me until the maintainers complete this list of release items that take some privilege.  If there is something else I should help with to complete this PR, just let me know. Otherwise I may go grab another easy starter.

**Comment by jcechace** [comment]
@dude0001  correct. I didn't get to it on Friday, however it will still require release so it needs to be done before we release 2.1.Alpha1.  Thanks for you contribution!

**Comment by dude0001** [comment]
Rebased on latest and updated md5 for the 2.1.0.Alpha1 release. I believe we just waiting on https://github.com/debezium/debezium/pull/4060.

**Comment by jpechane** [comment]
@dude0001 The other PR was merged, no we need to wait for Alpha2 that should be released at the end of this or at the begining of the next week.

File: connect/2.1/Dockerfile
Comment (jcechace) [inline connect/2.1/Dockerfile:27]: @dude0001 `io/debezium` shouldn't be here I think

**Review by jcechace — CHANGES_REQUESTED** [review]

**Comment by jcechace** [comment]
I took the librety of fixing the latest issues, merging now

**Pull request merged**
